src/main/java/com/facebook/openwifirrm/ucentral/models/AggregatedState.java

```diff
@@ -0,0 +1,214 @@
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package com.facebook.openwifirrm.ucentral.models;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import com.facebook.openwifirrm.ucentral.models.State.Interface.SSID.Association;
+import com.facebook.openwifirrm.ucentral.models.State.Interface.SSID.Association.Rate;
+
+public class AggregatedState {
+
+	public class AggregatedRate {
+		public long bitrate;
+		public int chwidth;
+		public List<Integer> mcs;
+
+		/** Constructor with no args */
+		public AggregatedRate() {
+			this.mcs = new ArrayList<>();
+		}
+
+		/** Constructor */
+		public AggregatedRate(List<Rate> rates) {
+			int size = rates.size();
+			if (size == 0) {
+				return;
+			}
+			add(rates);
+		}
+
+		/** Add a list of Rates to the AggregatedRate */
+		public void add(List<Rate> rates) {
+			if (rates.isEmpty()) {
+				return;
+			}
+			for (Rate rate : rates) {
+				this.mcs.add(rate.mcs);
+			}
+		}
+
+		/** Add a Rate to the AggregatedRate */
+		public void add(Rate rate) {
+			if (rate == null) {
+				return;
+			}
+			if (mcs == null || mcs.isEmpty()) {
+				this.bitrate = rate.bitrate;
+				this.chwidth = rate.chwidth;
+			}
+			this.mcs.add(rate.mcs);
+		}
+
+		/** Add an AggregatedRate with the same channel_width to the AggregatedRate */
+		public void add(AggregatedRate rate) {
+			if (rate.chwidth != chwidth) {
+				return;
+			}
+			if (mcs == null || mcs.isEmpty()) {
+				this.bitrate = rate.bitrate;
+				this.chwidth = rate.chwidth;
+			}
+			this.mcs.addAll(rate.mcs);
+		}
+	}
+
+	public String bssid;
+	public String station;
+	public long connected;
+	public long inactive;
+	public List<Integer> rssi;
+	public long rx_bytes;
+	public long rx_packets;
+	public AggregatedRate rx_rate;
+	public long tx_bytes;
+	public long tx_duration;
+	public long tx_failed;
+	public long tx_packets;
+	public AggregatedRate tx_rate;
+	public long tx_retries;
+	public int ack_signal;
+	public int ack_signal_avg;
+	public Radio radio;
+
+	public class Radio {
+		public int channel;
+		public int channel_width;
+		public int tx_power;
+
+		public Radio() {}
+
+		public Radio(int channel, int channel_width, int tx_power) {
+			this.channel = channel;
+			this.channel_width = channel_width;
+			this.tx_power = tx_power;
+		}
+
+		public Radio(int[] radios) {
+			this.channel = radios[0];
+			this.channel_width = radios[1];
+			this.tx_power = radios[2];
+		}
+
+		@Override
+		public int hashCode() {
+			return Objects.hash(channel, channel_width, tx_power);
+		}
+
+		@Override
+		public boolean equals(Object obj) {
+			if (this == obj) {
+				return true;
+			}
+			if (obj == null) {
+				return false;
+			}
+			if (getClass() != obj.getClass()) {
+				return false;
+			}
+
+			Radio other = (Radio) obj;
+			return channel == other.channel &&
+				channel_width == other.channel_width &&
+				tx_power == other.tx_power;
+		}
+	}
+
+	@Override
+	public int hashCode() {
```

Comment:
Add `equals()` or remove `hashCode()` (not a good idea to have one and not the other)